src/Type/BenevolentUnionType.php

```diff
@@ -42,4 +42,13 @@ public function isAcceptedBy(Type $acceptingType, bool $strictTypes): TrinaryLog
 		return TrinaryLogic::createNo();
 	}
 
+	/**
+	 * @param callable(Type $type): TrinaryLogic $getResult
+	 * @return TrinaryLogic
+	 */
+	protected function unionResults(callable $getResult): TrinaryLogic
+	{
+		return TrinaryLogic::maxMin(...array_map($getResult, $this->getTypes()));
```

Comment:
I'm not certain maxMin was the best method to use here, but I wasn't sure what the Maybe case...